hotel_dump.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+"""
+Demonstration of how to download a complete hotel-dump API dump.
+"""
+
+import hashlib
+import logging
+import os.path
+import re
+import sys
+
+import boto3
+
+
+def parse_args():
+    from argparse import ArgumentParser
+    argp = ArgumentParser(__doc__)
+    argp.add_argument("dest_folder", help="Existing empty folder where files should be stored")
+    return argp.parse_args()
+
+
+def md5(path):
+    """
+    Compute MD5 hash of file. Read it in chunks, so we don't need to fix the entire file in memory.
+    :param path: Path of file to hash.
+    :return: Hash value
+    """
+    hash_md5 = hashlib.md5()
+    with open(path, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+def download_latest(dest_folder):
+    """
+    Download the latest hotel-dump dump into a folder.
+    :param dest_folder: Path to destination folder on local disk.
+    """
+
+    logging.info("Connecting to Amazon S3")
+
+    # Connect to S3
+    s3 = boto3.resource("s3")
+
+    # Select the bucket containing dumps of the TrustYou API.
+    bucket = s3.Bucket("trustyou-api")
+
+    # The trustyou-api bucket contains snapshots of the TrustYou API at different points in time. They are organized in
+    # folders whose name is a timestamp.
+
+    logging.info("Looking for latest complete dump")
+
+    latest_date = None
+    for object_summary in bucket.objects.filter(Prefix="hotels/"):
+        done_file_match = re.match("^hotels/([^/]+)$", object_summary.key)
+        print(done_file_match)
+        if done_file_match:
+            date = done_file_match.group(1)
+            if latest_date is None or date > latest_date:
+                latest_date = date
+
+    assert latest_date is not None, "No complete dump folder found!"
+
+    logging.info("Downloading dump from %s", latest_date)
+
+    target_folder = os.path.join(dest_folder, "hotels", latest_date)
+    if not os.path.exists(target_folder):
+        os.makedirs(target_folder)
+
+    for object_summary in bucket.objects.filter(Prefix="hotels/{}".format(latest_date)):
```

Comment:
I don't think this loop makes sense, since you will always get exactly one match. You could instead extract the file name with the regex from line `56` and then download exactly that file (so basically continue continue with line 79)

Reply:
Hey I will review those and fix that for you